### PR TITLE
DataTable sorting accessibility improvements

### LINKED ISSUE #9127: Table Filters: : sorting feature is included in the TH, thus unnecessarily referenced by screen readers when navigating the table
Within a table that offers sorting functionality, the sorting functionality is included in the relevant column header. so, "sort column alphabetically" is in the column header. this is fine, but for screen reader users, every time the column header is referenced, the sort functionality should not be referenced. In other words, the sorting functionality should be offered in the column header data cell, but should not be referenced as part of the TH.
Here is a WET working example of where this issue occurs: https://wet-boew.github.io/v4.0-ci/demos/tables/tables-fltrs-en.html#dataset-filter2
And here is a Deque example of how to make it work so that the sort functionality is not referenced every time a screen reader navigates to the sortable column within the table.https://dequeuniversity.com/library/aria/table-sortable
https://dequeuniversity.com/library/aria/table-sortable

### PR DESCRIPTION
**Pull request by ghost**
## What does this pull request (PR) do? / Que fait cette demande « pull » (PR)?
A clear and concise description of what the code of this PR accomplishes. /// Une description claire et concise de ce que le code de cette PR accompli.

DataTable sorting accessibility improvements by preventing screen readers from indicating sort function on all cells.

## Additional information (optional) / Information additionnelle (optionel)

**General checklist / Liste de contrôle générale**
Make your own list for the purpose of your Pull request. /// Faites votre propre liste en fonction des besoins de votre demande « pull ».

- [ ] Ensure i18n is updated and bilingual

**Related issues / Requêtes associées**
List issues that are being closed or worked on with this pull request i.e. Closes #9127 /// Listez les autres requêtes (« issues » ou PR) qui sont fermées ou traitées avec cette demande de retrait ex. : Closes #9127

### CODE REVIEW

File: src/plugins/tables/_base.scss [hunk 1]
```diff
@@ -189,6 +189,18 @@ table {
 				}
 			}
 
+			button {
+				&.sorting-cnt {
+					-moz-appearance: none;
+					-webkit-appearance: none;
```
Comment (duboisp) [inline src/plugins/tables/_base.scss:195]: Usually the prefixed CSS are already managed by PostCSS during the build. Usually here we are using the naming from the W3C standard like "appearance".

File: src/plugins/tables/_base.scss [hunk 2]
```diff
@@ -189,6 +189,18 @@ table {
...
+					background: transparent;
+					border: 0;
+					font-family: inherit;
+					font-size: 100%;
```
Comment (duboisp) [inline src/plugins/tables/_base.scss:199]: Is this needed? Is there a value added to declare a font-size to 100%?

File: src/plugins/tables/tables.js [hunk 3]
```diff
@@ -84,7 +84,8 @@ var componentName = "wb-tables",
 				},
 				complete: function() {
 					var $elm = $( "#" + elmId ),
-						dataTableExt = $.fn.dataTableExt;
+						dataTableExt = $.fn.dataTableExt,
+						userDefaults = wb.getData( $elm, componentName );
```
Comment (duboisp) [inline src/plugins/tables/tables.js:88]: rename "userDefaults" for "settings". That is for consistency naming between plugin. For reference check the step 3 in the [hello world example](https://wet-boew.github.io/wet-boew/demos/helloworld/helloworld-en.html)

File: src/plugins/tables/tables.js [hunk 4]
```diff
@@ -114,11 +115,19 @@ var componentName = "wb-tables",
...
-					$elm.find( "th" ).append( "<span class='sorting-cnt'><span class='sorting-icons'></span></span>" );
+					// Handle sorting/ordering
+					var ordering = ( userDefaults && userDefaults.ordering === false ) ? false : true;
+					if ( ordering ) {
+						$elm.find( "th" ).each( function() {
+							var $th = $( this );
+							$th.html( "<button type='button' class='sorting-cnt' title='" + $th.html() + i18nText.aria.sortAscending + "'>" + $th.html() + " <span class='sorting-icons' aria-hidden='true'></span></button>" );
+							$th.removeAttr( "aria-label" );
+						} );
+						$elm.attr( "summary", "This table provides a sorting feature via the buttons across the column header row with only one instance visible at a time." );
```
Comment (duboisp) [inline src/plugins/tables/tables.js:126]: @GormFrank can you draft a French translation.

@thekodester We would might need to use an aria-label here considering the summary attribute is deprecated. For reference: https://html.spec.whatwg.org/multipage/tables.html#the-table-element and Obsolete - https://html.spec.whatwg.org/multipage/obsolete.html#HTMLTableElement-partial

**Review by duboisp — CHANGES_REQUESTED** [review]
See the requested change. 
Once completed we will perform a final test to see if it is working as expected.

Thanks @thekodester for the contribution.

**Comment by thekodester** [comment]
Someone we'll have to take over. I cannot put anymore time into this.

###### I am just a volunteer on this GitHub project.

**Comment by duboisp** [comment]
@thekodester no problem, thanks for your contribution.

@Ricokola will make the required changes on the top of your commits.

I am reopening this PR.

File: src/plugins/tables/_base.scss (hunk 2 above)
Comment (Ricokola) [reply src/plugins/tables/_base.scss:199]: After testing, it seems that removing  "font-size: 100%" as no effect on the content.

**Review by duboisp — CHANGES_REQUESTED** [review]
@Ricokola as discussed.

Please remove the focusability on the "th" and move the "aria-controls" on the button.

**Review by duboisp — APPROVED** [review]

**Review by GormFrank — APPROVED** [review]

**Pull request merged**